game-service/src/main/java/org/libertybikes/game/core/GameMap.java

```diff
@@ -0,0 +1,122 @@
+        movingObstacles.add(new MovingObstacle(size, size, GameBoard.BOARD_SIZE / 2 - size - offset, GameBoard.BOARD_SIZE / 2 - offset, -1, 0, speed));
+        obstacles.add(new Obstacle(size, size, GameBoard.BOARD_SIZE / 2 - offset, GameBoard.BOARD_SIZE / 2 - offset));
+        movingObstacles.add(new MovingObstacle(size, size, GameBoard.BOARD_SIZE / 2 + size - offset, GameBoard.BOARD_SIZE / 2 - offset, 1, 0, speed));
+        movingObstacles.add(new MovingObstacle(size, size, GameBoard.BOARD_SIZE / 2 - size - offset, GameBoard.BOARD_SIZE / 2 + size - offset, -1, 1, speed));
+        movingObstacles.add(new MovingObstacle(size, size, GameBoard.BOARD_SIZE / 2 - offset, GameBoard.BOARD_SIZE / 2 + size - offset, 0, 1, speed));
+        movingObstacles.add(new MovingObstacle(size, size, GameBoard.BOARD_SIZE / 2 + size - offset, GameBoard.BOARD_SIZE / 2 + size - offset, 1, 1, speed));
+
+    }
+
+    private void map4() {
+        //Smile
+        obstacles.add(new Obstacle(12, 1, (GameBoard.BOARD_SIZE / 10) * 3, GameBoard.BOARD_SIZE / 3));
+        obstacles.add(new Obstacle(12, 1, (GameBoard.BOARD_SIZE / 10) * 3, GameBoard.BOARD_SIZE / 3 + 12));
+        obstacles.add(new Obstacle(1, 11, (GameBoard.BOARD_SIZE / 10) * 3, GameBoard.BOARD_SIZE / 3 + 1));
+        obstacles.add(new Obstacle(1, 11, (GameBoard.BOARD_SIZE / 10) * 3 + 11, GameBoard.BOARD_SIZE / 3 + 1));
+        movingObstacles.add(new MovingObstacle(2, 2, (GameBoard.BOARD_SIZE / 10) * 3 + 8, GameBoard.BOARD_SIZE / 3 + 8, 1, 1));
+
+        obstacles.add(new Obstacle(12, 1, (GameBoard.BOARD_SIZE / 5) * 3, GameBoard.BOARD_SIZE / 3));
+        obstacles.add(new Obstacle(12, 1, (GameBoard.BOARD_SIZE / 5) * 3, GameBoard.BOARD_SIZE / 3 + 12));
+        obstacles.add(new Obstacle(1, 11, (GameBoard.BOARD_SIZE / 5) * 3, GameBoard.BOARD_SIZE / 3 + 1));
+        obstacles.add(new Obstacle(1, 11, (GameBoard.BOARD_SIZE / 5) * 3 + 11, GameBoard.BOARD_SIZE / 3 + 1));
+        movingObstacles.add(new MovingObstacle(2, 2, (GameBoard.BOARD_SIZE / 5) * 3 + 3, GameBoard.BOARD_SIZE / 3 + 8, -1, -1));
+
+        movingObstacles.add(new MovingObstacle(6, 6, (GameBoard.BOARD_SIZE / 2 - 3), GameBoard.BOARD_SIZE / 2, 1, 1));
+        obstacles.add(new Obstacle((GameBoard.BOARD_SIZE / 2), 1, GameBoard.BOARD_SIZE / 4, (GameBoard.BOARD_SIZE / 4) * 3 - 10));
+        for (int i = 0; i < 10; i++) {
+            obstacles.add(new Obstacle(1, 1, GameBoard.BOARD_SIZE / 4 - i, (GameBoard.BOARD_SIZE / 4) * 3 - i - 10));
+            obstacles.add(new Obstacle(1, 1, GameBoard.BOARD_SIZE / 4 + i + (GameBoard.BOARD_SIZE / 4) * 2, (GameBoard.BOARD_SIZE / 4) * 3 - i - 10));
+        }
+
+    }
+
+    public ArrayList<Obstacle> getObstacles() {
+        return obstacles;
+    }
+
+    public ArrayList<MovingObstacle> getMovingObstacles() {
+        return movingObstacles;
+    }
+
+    public int[][] getPlayerPositions() {
+        return playerPositions;
```

Comment:
instead of this returning null most of the time, can we eliminate `Player.PlayerStartingData` and make all of the map types return valid values?  Also, if this changed to `getPlayerPositionX(short playerNum)` and `getPlayerPositionY(short playerNum)` it would make working with the return value a bit easier